diff --git a/lib/cu_cp/cell_meas_manager/cell_meas_manager_helpers.cpp b/lib/cu_cp/cell_meas_manager/cell_meas_manager_helpers.cpp
index 1fa56b2b68..f02106fa09 100644
--- a/lib/cu_cp/cell_meas_manager/cell_meas_manager_helpers.cpp
+++ b/lib/cu_cp/cell_meas_manager/cell_meas_manager_helpers.cpp
@@ -239,7 +239,7 @@ rrc_meas_obj_nr ocudu::ocucp::generate_measurement_object(const serving_cell_mea
   meas_obj_nr.smtc1                  = cfg.ssb_mtc;
 
   // Mandatory fields.
-  meas_obj_nr.ref_sig_cfg.ssb_cfg_mob.emplace().derive_ssb_idx_from_cell = true;
+  meas_obj_nr.ref_sig_cfg.ssb_cfg_mob.emplace().derive_ssb_idx_from_cell = false;
   meas_obj_nr.nrof_ss_blocks_to_average.emplace()                        = 8; // TODO: remove hardcoded values
   meas_obj_nr.quant_cfg_idx                                              = 1; // TODO: remove hardcoded values
   meas_obj_nr.freq_band_ind_nr.emplace()                                 = nr_band_to_uint(cfg.band.value());
diff --git a/lib/ngap/procedures/ngap_handover_resource_allocation_procedure.cpp b/lib/ngap/procedures/ngap_handover_resource_allocation_procedure.cpp
index 35078fe8f0..af09fd38bc 100644
--- a/lib/ngap/procedures/ngap_handover_resource_allocation_procedure.cpp
+++ b/lib/ngap/procedures/ngap_handover_resource_allocation_procedure.cpp
@@ -45,6 +45,7 @@ void ngap_handover_resource_allocation_procedure::operator()(coro_context<async_
       // Update UE with AMF UE ID.
       ngap_ue_context& ue_ctxt = ue_ctxt_list[ho_ack.ue_index];
       ue_ctxt_list.update_amf_ue_id(ue_ctxt.ue_ids.ran_ue_id, amf_ue_id);
+      ue_ctxt.serving_guami = request.guami;
 
       // Store Core Network Assist Info for Inactive if present.
       if (request.core_network_assist_info_for_inactive.has_value()) {
